R/zchunk_LA115.roofPV.R

```diff
@@ -29,42 +27,45 @@ module_energy_LA115.roofPV_DISABLED <- function(command, ...) {
     # Load required inputs
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
     A15.roofPV_curves <- get_data(all_data, "energy/A15.roofPV_curves")
-    A15.roofPV_TechChange <- get_data(all_data, "energy/A15.roofPV_TechChange")
-    L100.Pop_thous_ctry_Yh <- get_data(all_data, "L100.Pop_thous_ctry_Yh")
+    L100.Pop_thous_ctry_Yh <- get_data(all_data, "temp-data-inject/L100.Pop_thous_ctry_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Categorizing countries and regions to GCAM Region ID; selecting population in 2010
+    L100.Pop_thous_ctry_Yh %>%
+      gather(year, value, X1700:X2010) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) %>%
+      filter(year == "2010") %>%
+      left_join(iso_GCAM_regID, by = "iso") %>%
+      select(-iso, -country_name, -year) ->
+      x
+
+    # Sum of population by regions
+    x %>%
+      group_by(region_GCAM3) %>%
+      summarise(popSum = sum(value)) ->
+      pop_RG3
+
+    # Building resource curves by GCAM Region ID. Because some regions span multiple GCAM Region IDs, population in 2010 is used to allocate.
+    x %>%
+      left_join(pop_RG3, by = "region_GCAM3") %>%
+      left_join(A15.roofPV_curves, by = "region_GCAM3") %>%
+      mutate(maxSubResource = maxSubResource*value/popSum) %>%
```

Comment:
Is it possible to use `left_join_error_no_match` in one or both of the joins above? If so, please do.